core/src/test/java/com/linecorp/armeria/client/HttpClientFactoryTest.java

```diff
@@ -103,4 +114,35 @@ protected HttpResponse doGet(ServiceRequestContext ctx,
             });
         }
     }
+
+    @Test
+    void useOpenSsl() throws ExecutionException, InterruptedException {
+        try (ClientFactory sslEnabledClient = ClientFactory.builder().tlsNoVerify().build();
+             ClientFactory sslDisabledClient = ClientFactory.builder().useOpenSsl(false).build();
+        ) {
+            assertThat(sslEnabledClient.options().useOpenSsl()).isTrue();
+            assertThat(sslDisabledClient.options().useOpenSsl()).isFalse();
+
+            final AggregatedHttpResponse response2 = WebClient.builder(SessionProtocol.HTTPS,
+                                                                       server.httpsEndpoint())
+                                                              .factory(sslEnabledClient).build()
+                                                              .get("/hello").aggregate().join();
+            assertThat(response2.status()).isEqualTo(HttpStatus.OK);
+            assertThat(response2.contentUtf8()).isEqualTo("hello");
+
+            final AggregatedHttpResponse response3 = WebClient.builder(SessionProtocol.HTTP,
+                                                                       server.httpEndpoint())
+                                                              .factory(sslDisabledClient).build()
+                                                              .get("/hello").aggregate().join();
+            assertThat(response3.status()).isEqualTo(HttpStatus.OK);
+            assertThat(response3.contentUtf8()).isEqualTo("hello");
+
+            assertThatThrownBy(() -> WebClient.builder(SessionProtocol.HTTPS, server.httpsEndpoint())
+                                              .factory(sslDisabledClient).build()
+                                              .get("/hello").aggregate().join())
+                    .isInstanceOfSatisfying(CompletionException.class, e -> {
+                        assertThat(e).hasCauseInstanceOf(UnprocessedRequestException.class);
+                    });
```

Comment:
I'm not sure what these lines are testing 😅   Can we just test once with `tlsVerify` disabled for each ClientFactory?

```
ClientFactory sslDisabledClient = ClientFactory.builder().useOpenSsl(false).tlsNoVerify().build();
...
final AggregatedHttpResponse response3 = WebClient.builder(SessionProtocol.HTTPS,
                                                           server.httpsEndpoint())
                                                  .factory(sslDisabledClient).build()
                                                  .get("/hello").aggregate().join();
assertThat(response3.status()).isEqualTo(HttpStatus.OK);
assertThat(response3.contentUtf8()).isEqualTo("hello");
```

Reply:
Great! It would be more clear!

Reply:
If OpenSSL is disabled, the JDK SSL engine is used instead.

Reply:
Oh, then it seems that above mentioned scenario isn't valid. I'll remove the test 👍